### DISCUSSION TITLE
Run Ci build test on every commit in a PR

### DISCUSSION THREAD

**Discussion by axboe** [axboe/liburing#1227] [Q&A] [answered]
One of the things that liburing tries to do is retain bisectability for all commits in the tree. That means that each checkout should build. However, our CI testing only checks the final commit/merge, not every commit in the pull request. I'm not very CI savvy, but here's an example of someone that DOES check every commit:

https://github.com/landlock-lsm/rust-landlock/blob/main/.github/workflows/rust.yml

if someone would be interested in helping out with this, I'd love to see it for liburing.

**Comment by ammarfaizi2** [comment, marked as answer]
I am working on it.

Once it's complete, I expect you to maintain a `next` branch where you rebase whenever a build error is found. This ensures that only clean, fast-forward commits are merged into `master`.
